tracee-rules/engine/engine.go

```diff
@@ -62,24 +67,34 @@ func NewEngine(sigs []types.Signature, sources EventSources, output chan types.F
 	return &engine
 }
 
+// signatureStartFunc is the signature handling business logics.
+func (engine *Engine) signatureStartFunc(signature types.Signature, c chan types.Event) {
+	for {
+		select {
+		case e, ok := <-c:
+			if !ok {
+				// channel closed
+				return
+			}
```

Comment:
why did you prefer this over the current code for `e:= range c` which automatically handles channel close?